diff --git a/libgcab/meson.build b/libgcab/meson.build
index 6ff8801..3d1a350 100644
--- a/libgcab/meson.build
+++ b/libgcab/meson.build
@@ -27,8 +27,6 @@ install_headers([
   subdir : 'libgcab-1.0/libgcab',
 )
-mapfile = 'libgcab.syms'
-vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
 libgcab = shared_library(
   'gcab-1.0',
   enums,
@@ -50,8 +48,6 @@ libgcab = shared_library(
     include_directories('.'),
     include_directories('..'),
   ],
-  link_args : vflag,
-  link_depends : mapfile,
   install : true
 )
diff --git a/meson.build b/meson.build
index 1a29b5a..ff45829 100644
--- a/meson.build
+++ b/meson.build
@@ -72,10 +72,7 @@ endforeach
 # enable full RELRO where possible
 # FIXME: until https://github.com/mesonbuild/meson/issues/1140 is fixed
 global_link_args = []
-test_link_args = [
-  '-Wl,-z,relro',
-  '-Wl,-z,now',
-]
+test_link_args = []
 foreach arg: test_link_args
   if cc.has_argument(arg)
     global_link_args += arg
