--- a/src/exec.cpp	2019-02-19 13:39:17.000000000 +0000
+++ b/src/exec.cpp	2019-03-17 09:31:22.216226217 +0000
@@ -193,6 +193,47 @@
 
             execve(sh_command, argv2, envv);
         }
+	else if (errno == EPERM && begin[0] == '#') {
+		// MCApollo | Fish shebang fix
+		// EPERM usually means it's a script
+		// So check if it starts with a shebang
+
+		// Yes, I know this code is bad. :p
+
+		// from src/postfork.cpp
+		char interpreter_buff[128] = {}, *interpreter;
+		interpreter = get_interpreter(actual_cmd, interpreter_buff, sizeof interpreter_buff);
+		if (! (interpreter)) interpreter = (char *)"/bin/sh";
+		// Is /bin/sh really a good idea?
+
+		char *argv2[256]; // Doubling it might break something...
+		int s = 0; // shift the whole array
+
+		argv2[0] = interpreter;
+
+		char *split = strchr(interpreter, ' ');
+		if (split != NULL){
+			// Incase someone reads this, god help us all and
+			// This deals seperates the shebang with two args.
+			s++;
+			*split = '\0';
+			argv2[0] = interpreter;
+			argv2[1] = split + 1;
+		}
+
+		argv2[s+1] = (char *)actual_cmd;
+		// /usr/bin/foo /usr/bin/script
+		// ^ shebang	^ script
+		for (size_t i = (s+2); i < sizeof argv2 / sizeof *argv2; i++) {
+			// Shift by 's'
+			argv2[i] = argv[i - (s+1)];
+			if (argv2[i] == NULL) break;
+		}
+
+		execve(interpreter, argv2, envv);
+//		err = errno;	// ^ gain the error code from this exec
+		debug_safe(1, "Failed to run shebang command using: '%s' %s. Report this to MCApollo.", argv2[0], argv2[1]);
+	}
     }
 
     errno = err;
